_plugins/search-indexer.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require "jekyll/hooks"
+require "jekyll/document"
+require "json"
+
+##
+# This singleton facilitates production of an indexable JSON representation of the content to populate a data source
+# to provide search functionality.
+module Jekyll::ContentIndexer
+
+  ##
+  # The collection that will get stores as the output
+  @data = []
+
+  ##
+  # Pattern to identify documents that should be excluded based on their URL
+  @excluded_paths = /(^\/blog\/(page\d|$)|^\/events\/([^\d]|$)|^\/faqs|\.(css|js|json|map|xml|txt|yml)$|\/404\.html)/i.freeze
+
+  ##
+  # Pattern to identify block HTML tags (not comprehensive)
+  @html_block_tags = /\s*<[?\/]?(article|blockquote|d[dlt]|div|fieldset|form|h|li|main|nav|[ou]l|p|section|table|t[rd]).*?>\s*/im.freeze
+
+  ##
+  # Pattern to identify certain HTML tags whose content should be excluded from indexing
+  @html_excluded_tags = /\s*<(head|style|script|h1).*?>.*?<\/\1>/im.freeze
+
+  ##
+  # Pattern to extract the page heading from div.copy-banner > div.container > h1 > a
+  @header_matcher = /<div.+?class="[^"]*copy-banner[^"]*".*?>.*?<div.+?class="[^"]*container[^"]*".*?>.*?<h1.*?><a.*?>\s*([^<]+)\s*<\/a>.*?<\/h1>/m.freeze;
+
+  ##
+  # Initializes the singleton by recording the site
+  def self.init(site)
+    @site = site
+  end
+
+  ##
+  # Processes a Document or Page and adds it to the collection
+  def self.add(page)
+    return if @excluded_paths.match(page.url)
+    return if page.data['omit_from_search']
```

Comment:
does `page` also work for layouts? So, say, I want any page with a given layout to _not_ show up in search. Would this work?

Reply:
`page.data` is populated the particular document's front matter after its content has been converted but before before being rendered in a layout. Hence, setting `omit_from_search: true` will not work. However, `@excluded_paths` is a hard-coded regex and can be used for this purpose.

Reply:
OK. In the templating language for Jekyll, `page` only refers to front matter directly on the page but there are similar objects for layouts (`layout`). I'll experiment at a later - specifying it up the stream might be very useful.

Reply:
The hook being used is `post_convert` to avoid having to remove content added by layouts. I might be able to look check `omit_from_search` after rendering and remove documents from the index. I will experiment with that for the next iteration.